### PR TITLE
Fix bug where long branch names overflow list view

### LINKED ISSUE #5625: Long branch name overflows in dashboard
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

If a branch name is long, it overflows adjacent repositories in the dashboard:

image

**What is the expected behavior?**

Text is truncated with ellipsis or similar.

**Environment you encounter the issue:**

- GitExtensions version: beta1

### PR DESCRIPTION
**Pull request by anaplian**
Fixes #5625 

Changes proposed in this pull request:
 - Include icon width when calculating tile size
 - Add ellipsis when truncating branch names/repo paths
 
Screenshots:

Before:

gitextensions-issue

After:

gitextensions-fixed

What did I do to test the code and ensure quality:
- Ran the tests
- Ran the linter

Has been tested on (remove any that don't apply):
- GIT 2.19
- Windows 7

### CODE REVIEW

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs [hunk 1]
```diff
@@ -414,9 +414,15 @@ private static void RepositoryContextAction(ToolStripItem menuItem, Action<Repos
...
+            var width = TextRenderer.MeasureText(text, font).Width;
+            if (width < maxWidth)
+            {
+               return text;
+            }
+
             while (text.Length > 1)
             {
-                var width = TextRenderer.MeasureText(text, font).Width;
+                width = TextRenderer.MeasureText(text + "...", font).Width;
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs]: This should be elipsis character (inconsistently used in GE though)

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs [hunk 2]
```diff
@@ -414,9 +414,16 @@ private static void RepositoryContextAction(ToolStripItem menuItem, Action<Repos
 
         private static string ShortenText(string text, Font font, float maxWidth)
         {
+            var ellipsis = '…';
+            var width = TextRenderer.MeasureText(text, font).Width;
+            if (width < maxWidth)
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs]: This should be `<=`, shouldn't it?

Comment (drewnoakes) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs]: I think `<` is fine. Possibly even `< maxWidth - somePadding`.

**Comment by RussKie** [comment]
Thank you

**Comment by drewnoakes** [comment]
Thanks @anaplian!

**Pull request merged**